src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -199,6 +208,7 @@ public final void traverse() {
             // Since reads in each shard are lazily fetched, we need to pass the filter to the window
             // instead of filtering the reads directly here
             readShard.setReadFilter(countedFilter);
+            readShard.setDownsampler(maxReadsPerAlignmentStart > 0 ? new PositionalDownsampler(maxReadsPerAlignmentStart, getHeaderForReads()) : null);
```

Comment:
There would still be overhead, since you'd have to wrap the no-op downsampler in a `ReadsDownsamplingIterator`. It seems silly to do this, and is not really "cleaner" or more readable.